shared/bitutil/bit.go

```diff
@@ -83,3 +78,15 @@ func fillBit(target byte, index uint64) byte {
 	bitShift := 7 - index
 	return target ^ (1 << bitShift)
 }
+
+// BitfieldBit Extract the bit in `bitfield` at position `i`.
+// Spec pseudocode definition:
+//   def get_bitfield_bit(bitfield: bytes, i: int) -> int:
+//     """
+//     Extract the bit in ``bitfield`` at position ``i``.
+//     """
+//     return (bitfield[i // 8] >> (i % 8)) % 2
+func BitfieldBit(bitfield []byte, i int) byte {
+	by := (bitfield[i>>3] >> (uint(i) & 0x7)) & 0x1
```

Comment:
maybe make a few of these expressions vars ? it will be easier to read. Putting everything in oneline makes
it difficult to understand what is going on

Reply:
made it readable

Reply:
thanks